repair/row_level.cc

```diff
@@ -586,9 +587,11 @@ class repair_writer : public enable_lw_shared_from_this<repair_writer> {
         _mq = std::move(queue_handle);
         auto writer = shared_from_this();
         _writer_done = mutation_writer::distribute_reader_and_consume_on_shards(_schema, std::move(queue_reader),
-                [&db, &sys_dist_ks, &view_update_gen, reason = this->_reason, estimated_partitions = this->_estimated_partitions] (flat_mutation_reader reader) {
-            auto& t = db.local().find_column_family(reader.schema());
-            return db::view::check_needs_view_update_path(sys_dist_ks.local(), t, reason).then([t = t.shared_from_this(), estimated_partitions, reader = std::move(reader), reason, &view_update_gen] (bool use_view_update_path) mutable {
+                [&db, &sys_dist_ks, &view_update_gen, reason = this->_reason, estimated_partitions = this->_estimated_partitions] (flat_mutation_reader reader) -> future<> {
```

Comment:
I think this is broken. Lambda coroutines are captured by reference, and create_writer() is not called in a thread or is a coroutine itself.

Reply:
Let's split the coroutinization into its own patch, it is delicate.

Reply:
It passes debug unit tests for me, but I'm not sure they cover repair very well.
I'll look into extracting the lambda function it into a method, it will probably also simplify things a bit.

Reply:
@avikivity fixed in v2.

Reply:
unit tests don't repair at all. I hope to improve this with seastar support for multiple instances and @xemul war on globals.